modules/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3ServiceTests.java

```diff
@@ -47,19 +48,36 @@ public void testRetryOn403RetryPolicy() {
         e.setStatusCode(403);
         e.setErrorCode("InvalidAccessKeyId");
 
-        // Retry on 403 invalid access key id
+        // AWS default retry condition does not retry on 403
+        assertFalse(PredefinedRetryPolicies.DEFAULT_RETRY_CONDITION.shouldRetry(mock(AmazonWebServiceRequest.class), e, between(0, 9)));
+
+        // The retryable 403 condition retries on 403 invalid access key id
         assertTrue(
             S3Service.RETRYABLE_403_RETRY_POLICY.getRetryCondition().shouldRetry(mock(AmazonWebServiceRequest.class), e, between(0, 9))
         );
 
-        // Not retry if not 403 or not invalid access key id
         if (randomBoolean()) {
+            // Random for another error status that is not 403
             e.setStatusCode(randomValueOtherThan(403, () -> between(0, 600)));
+            // Retryable 403 condition delegates to the AWS default retry condition. Its result must be consistent with the decision
+            // by the AWS default, e.g. some error status like 429 is retryable by default, the retryable 403 condition respects it.
+            if (PredefinedRetryPolicies.DEFAULT_RETRY_CONDITION.shouldRetry(mock(AmazonWebServiceRequest.class), e, between(0, 9))) {
+                assertTrue(
+                    S3Service.RETRYABLE_403_RETRY_POLICY.getRetryCondition()
+                        .shouldRetry(mock(AmazonWebServiceRequest.class), e, between(0, 9))
+                );
+            } else {
+                assertFalse(
+                    S3Service.RETRYABLE_403_RETRY_POLICY.getRetryCondition()
+                        .shouldRetry(mock(AmazonWebServiceRequest.class), e, between(0, 9))
+                );
+            }
```

Comment:
I would find something like this easier to read:

```suggestion
            boolean actual = S3Service.RETRYABLE_403_RETRY_POLICY.getRetryCondition()
                        .shouldRetry(mock(AmazonWebServiceRequest.class), e, between(0, 9));
            boolean expected  = PredefinedRetryPolicies.DEFAULT_RETRY_CONDITION.shouldRetry(mock(AmazonWebServiceRequest.class), e, between(0, 9));
            assertThat(actual, equals(expected));
```